src/shared/local-addresses.c

```diff
@@ -93,6 +91,11 @@ int local_addresses(sd_netlink *context, int ifindex, int af, struct local_addre
                 if (flags & IFA_F_DEPRECATED)
                         continue;
 
+                if (!ret) {
+                        n_list++;
+                        continue;
+                }
+
```

Comment:
We have several checks below. I think it is better to not skip them.

BTW, this change is not necessary for the next commit. I think it is ok to drop this change.

Reply:
indeed. this makes no sense. will drop.